src/components/currencies/currencyList.js

```diff
@@ -0,0 +1,42 @@
+import { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import {
+  clearCountryCode,
+  fetchAllCurrencies,
+} from '../../redux/currency/currencySlice';
+import CurrencyCard from './currencyCard';
+
+function CurrencyList() {
+  const currencyData = useSelector((state) => state.currency);
+  const dispatch = useDispatch();
+  useEffect(() => {
+    dispatch(fetchAllCurrencies());
+    dispatch(clearCountryCode());
+  }, [dispatch]);
+
+  if (currencyData.isLoading) return <p className="status">Loading...</p>;
+  if (currencyData.isError) {
+    return (
+      <p className="status" data-testid="error-msg">
+        {currencyData.isError}
+      </p>
+    );
+  }
+  const arr = Object.entries(currencyData.currenciesData).slice(3, 11);
+  return (
+    <section className="currencies">
+      <div className="overview">
+        <div />
+        <h3>World currencies names</h3>
+      </div>
+      <h1 className="title">Stats by currency</h1>
+      <ul className="countryList">
+        {arr.map((el) => (
+          <CurrencyCard key={el[0]} curCode={el[0]} curName={el[1]} />
+        ))}
+      </ul>
+    </section>
```

Comment:
- [ ] Hey, great job with the project 😄 , I admire your effort so far, You've done well. I just wanted to point out something concerning the search. As it's a requirement to add a search functionality so a user could search by currency name, I don't see it working as intended. when a user searches for a currency, it should show up what matches the search  Please, try to add this implementation to meet this part of the project requirement. 👍

### Neslon's design

![Capture](https://github.com/hmunish/react-metrics/assets/80458199/3804c445-839a-4870-8906-febe2875dfbd)